core/arch/proc_api.h

```diff
@@ -158,7 +159,16 @@ typedef struct {
     uint ext_flags_ecx;  /**< extended feature flags stored in ecx */
     uint sext_flags_ebx; /**< structured extended feature flags stored in ebx */
 } features_t;
+#elif defined(AARCHXX)
+typedef struct {
+    uint64 flags_aa64isar0; /**< feature flags stored in ID_AA64ISAR0_EL1 */
+    uint64 flags_aa64isar1; /**< feature flags stored in ID_AA64ISAR1_EL1 */
+    uint64 flags_aa64pfr0;  /**< feature flags stored in ID_AA64PFR0_EL1 */
+} features_t;
+typedef enum { AA64ISAR0 = 0, AA64ISAR1 = 1, AA64PFR0 = 2 } feature_reg_idx_t;
```

Comment:
If this is not exported, best to move to a private header or impl file and not keep it in an _api.h file.  If it is exported (doesn't look like it), it needs doxygen docs for the enum and each member.

Reply:
I don't know how to do this without #including a private header in proc_api.h
Any advice?

Reply:
Looking more closely: ok, you mean that this is used directly in the exposed enum values below, as in ` FEATURE_AES = DEF_FEAT(AA64ISAR0, 1, 1, 0), `.  I guess it does need to be here then.  What does the generated doxygen look like?  I guess it has DEF_FEAT and AA64ISAR0 but they aren't links.

Reply:
One option would be to doxygen-document them so the user can see them in the html.  Or it doesn't seem that bad to have them not documented: can live w/ the negative of not having the html show the components and could possibly change them later.

Reply:
> What does the generated doxygen look like? I guess it has DEF_FEAT and AA64ISAR0 but they aren't links.

I can't see AArch64 `feature_bit_t` output in the HTML.

I built with
`cmake -DDEBUG=ON -DBUILD_TESTS=ON -DINTERNAL=ON -DBUILD_DOCS=ON`
and the X86 enum `feature_bit_t` list appeared in full but no AArch64.

Have I built it wrong? Or is there some Doxygen/C pre-processor magic which needs doing?

Reply:
I think it's the use of `#elif`.  See `PREDEFINED` in API.doxy where it has all OS and arch defines set: so if avoid else in the _api.h headers we get it all in any build.

I see at least one comment about this:
```
core/ir/instr_api.h:/* We resist using #elif here because otherwise doxygen will be unable to
```

But I see a bunch of other uses of elses: what we need is a lint rule on this. I filed #5496 on this.

Reply:
Removing the `#elif` does result in both X86 and AArch64 documentation being generated but it merges the X86 and AArch64 HTML comments into single blocks. It also confuses FEATURE_AES because it's common to both. I've renamed the AArch64 version to FEATURE_AESX.

I've re-worded the Doxygen comments to make more sense when rendered to HTML as well as adding X86 and AArch64 labels wherever needed to make it clearer for users. Not ideal but I think will do for now.

> Or it doesn't seem that bad to have them not documented: can live w/ the negative of not having the html show the components and could possibly change them later.

Yes, I would prefer to wait.